src/runtime/hexagon_remote/halide_remote.cpp

```diff
@@ -200,7 +229,30 @@ int halide_hexagon_remote_power_hvx_on() {
 int halide_hexagon_remote_power_hvx_off() {
     power_ref_count--;
     if (power_ref_count == 0) {
-        HAP_power_request(0, 0, -1);
+        HAP_power_request_t request;
+
+        request.type = HAP_power_set_HVX;
+        request.hvx.power_up = FALSE;
+        int retval = HAP_power_set(NULL, &request);
```

Comment:
It does pass that check.

I guess I'm still suspicious that these APIs are actually doing anything because the take the same amount of time regardless of whether Hexagon was previously powered up/clocked up or not. I also added a call to qurt_hvx_lock inside power_hvx_on to make sure that it wasn't asynchronously powering HVX on (and blocking upon call to qurt_hxv_lock), but I still see the same ~3-5 ms regardless of whether HVX is previously powered on or not.